graph/run-dynamic.go

```diff
@@ -299,7 +299,6 @@ func ConvergeDynamicThread[EP EPP[E], V VPI[V], E EPI[E], M MVI[M], N any, A Alg
 			}
 			epoch = false
 			completed = false
-			g.TerminateVotes[tidx] = 0
```

Comment:
Other threads might have terminated again before this thread, which potentially can carry messages, gets to run again. This is more likely to happen when the graph is still very small.